### ISSUE TITLE
Kernel 4.13.6 and steam glxchooseVisual Error

### ISSUE THREAD

**Issue by trite2k3** [ValveSoftware/steam-for-linux#5205]
slackware64-current

Basically installed like this:
https://forums-archive.eveonline.com/default.aspx?g=posts&t=502364

fresh install of slackware 64-current today with latest nvidia installed from nvidia.com .run-file and multilib from alien slakcware repo

ran glxgears and moved the windows like crazy, stable fps and everything seems to work. got GSYNC, two 1440p monitors and GSYNC on the main one.

restarted multiple times
steam doesnt start

root@bifrost:~# nvidia-smi -a

==============NVSMI LOG==============

Timestamp : Fri Oct 13 19:31:02 2017
Driver Version : 384.90

Attached GPUs : 1
GPU 00000000:01:00.0
Product Name : GeForce GTX 980
Product Brand : GeForce
Display Mode : Enabled
Display Active : Enabled
Persistence Mode : Disabled
Accounting Mode : Disabled
Accounting Mode Buffer Size : 1920
Driver Model
Current : N/A
Pending : N/A
Serial Number : 0323614114278
GPU UUID : GPU-5b953fca-b3dc-b086-25c3-7f246f4cd127
Minor Number : 0
VBIOS Version : 84.04.1F.00.02
MultiGPU Board : No
Board ID : 0x100
GPU Part Number : N/A
Inforom Version
Image Version : G001.0000.01.03
OEM Object : 1.1
ECC Object : N/A
Power Management Object : N/A
GPU Operation Mode
Current : N/A
Pending : N/A
GPU Virtualization Mode
Virtualization mode : None
PCI
Bus : 0x01
Device : 0x00
Domain : 0x0000
Device Id : 0x13C010DE
Bus Id : 00000000:01:00.0
Sub System Id : 0x85041043
GPU Link Info
PCIe Generation
Max : 3
Current : 3
Link Width
Max : 16x
Current : 16x
Bridge Chip
Type : N/A
Firmware : N/A
Replays since reset : 0
Tx Throughput : 0 KB/s
Rx Throughput : 8000 KB/s
Fan Speed : 30 %
Performance State : P0
Clocks Throttle Reasons
Idle : Not Active
Applications Clocks Setting : Not Active
SW Power Cap : Not Active
HW Slowdown : Not Active
Sync Boost : Not Active
SW Thermal Slowdown : Not Active
FB Memory Usage
Total : 4035 MiB
Used : 193 MiB
Free : 3842 MiB
BAR1 Memory Usage
Total : 256 MiB
Used : 4 MiB
Free : 252 MiB
Compute Mode : Default
Utilization
Gpu : 2 %
Memory : 2 %
Encoder : 0 %
Decoder : 0 %
Encoder Stats
Active Sessions : 0
Average FPS : 0
Average Latency : 0
Ecc Mode
Current : N/A
Pending : N/A
ECC Errors
Volatile
Single Bit
Device Memory : N/A
Register File : N/A
L1 Cache : N/A
L2 Cache : N/A
Texture Memory : N/A
Texture Shared : N/A
CBU : N/A
Total : N/A
Double Bit
Device Memory : N/A
Register File : N/A
L1 Cache : N/A
L2 Cache : N/A
Texture Memory : N/A
Texture Shared : N/A
CBU : N/A
Total : N/A
Aggregate
Single Bit
Device Memory : N/A
Register File : N/A
L1 Cache : N/A
L2 Cache : N/A
Texture Memory : N/A
Texture Shared : N/A
CBU : N/A
Total : N/A
Double Bit
Device Memory : N/A
Register File : N/A
L1 Cache : N/A
L2 Cache : N/A
Texture Memory : N/A
Texture Shared : N/A
CBU : N/A
Total : N/A
Retired Pages
Single Bit ECC : N/A
Double Bit ECC : N/A
Pending : N/A
Temperature
GPU Current Temp : 49 C
GPU Shutdown Temp : 96 C
GPU Slowdown Temp : 91 C
GPU Max Operating Temp : N/A
Memory Current Temp : N/A
Memory Max Operating Temp : N/A
Power Readings
Power Management : Supported
Power Draw : 45.13 W
Power Limit : 180.00 W
Default Power Limit : 180.00 W
Enforced Power Limit : 180.00 W
Min Power Limit : 120.00 W
Max Power Limit : 225.00 W
Clocks
Graphics : 911 MHz
SM : 911 MHz
Memory : 3505 MHz
Video : 838 MHz
Applications Clocks
Graphics : 1126 MHz
Memory : 3505 MHz
Default Applications Clocks
Graphics : 1126 MHz
Memory : 3505 MHz
Max Clocks
Graphics : 1392 MHz
SM : 1392 MHz
Memory : 3505 MHz
Video : 1281 MHz
Max Customer Boost Clocks
Graphics : N/A
Clock Policy
Auto Boost : N/A
Auto Boost Default : N/A
Processes
Process ID : 1334
Type : G
Name : /usr/libexec/Xorg
Used GPU Memory : 190 MiB

root@bifrost:# lspci -vnnn | perl -lne 'print if /^\d+:.+([\S+:\S+])/' | grep VGA
00:02.0 VGA compatible controller [0300]: Intel Corporation Xeon E3-1200 v3/4th Gen Core Processor Integrated Graphics Controller [8086:0412] (rev 06) (prog-if 00 [VGA controller])
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation GM204 [GeForce GTX 980] [10de:13c0] (rev a1) (prog-if 00 [VGA controller])
root@bifrost:# nvidia-settings &
[1] 2496
root@bifrost:# glxgears
Running synchronized to the vertical refresh. The framerate should be
approximately the same as the monitor refresh rate.
719 frames in 5.0 seconds = 143.759 FPS
720 frames in 5.0 seconds = 143.808 FPS
720 frames in 5.0 seconds = 143.803 FPS
720 frames in 5.0 seconds = 143.810 FPS
720 frames in 5.0 seconds = 143.806 FPS
720 frames in 5.0 seconds = 143.807 FPS
719 frames in 5.0 seconds = 143.606 FPS
720 frames in 5.0 seconds = 143.806 FPS
720 frames in 5.0 seconds = 143.807 FPS
720 frames in 5.0 seconds = 143.806 FPS
720 frames in 5.0 seconds = 143.807 FPS
720 frames in 5.0 seconds = 143.806 FPS
720 frames in 5.0 seconds = 143.806 FPS
720 frames in 5.0 seconds = 143.808 FPS
720 frames in 5.0 seconds = 143.805 FPS
720 frames in 5.0 seconds = 143.809 FPS
720 frames in 5.0 seconds = 143.807 FPS
XIO: fatal IO error 11 (Resource temporarily unavailable) on X server ":0.0"
after 60 requests (60 known processed) with 0 events remaining.
root@bifrost:# uname -a
Linux bifrost 4.13.6 #1 SMP Fri Oct 13 18:27:00 CEST 2017 x86_64 Intel(R) Core(TM) i7-4770K CPU @ 3.50GHz GenuineIntel GNU/Linux
[1]+ Done nvidia-settings

root@bifrost:~# nvidia-smi
Fri Oct 13 19:54:54 2017       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 384.90                 Driver Version: 384.90                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 980     Off  | 00000000:01:00.0  On |                  N/A |
| 30%   49C    P0    45W / 180W |    193MiB /  4035MiB |      1%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1334      G   /usr/libexec/Xorg                            190MiB |
+-----------------------------------------------------------------------------+

**Comment by kisak-valve** [comment]
Hello @trite2k3, can you confirm any 32 bit opengl application works? The most common cause of this kind of issue is that the 32 bit opengl stack is not sane. For the nvidia driver, the userspace library version must match the running kernel module's version.

**Comment by trite2k3** [comment]
root@bifrost:~# /usr/bin/32/glxgears
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
Error: couldn't get an RGB, Double-buffered visual

I'm going to try and troubleshoot it.

**Comment by trite2k3** [comment]
Reinstalled nvidia drivers after enabling/installing multilib. thanks and sorry, maybe someone else find this post useful.

root@bifrost:~# /usr/bin/32/glxgears
Running synchronized to the vertical refresh.  The framerate should be
approximately the same as the monitor refresh rate.
721 frames in 5.0 seconds = 144.005 FPS
XIO:  fatal IO error 11 (Resource temporarily unavailable) on X server ":0.0"
      after 58 requests (58 known processed) with 0 events remaining.
root@bifrost:~#

**Comment by trite2k3** [comment]
As additional information, I found out that I had accidently slackware64-14.2 system upgrade /etc/slackpkg/mirrorlist was pointing at 14.2.

I repointed it at slackware64-current instead and remade full system upgrade and installed multilib again, then this problem reappeared. And it later went away after installing NVIDIA*.run again.

So I think this happens each time I install multilib after NVIDIA*.run.
